app/upgrades/v2_5_0/v2_5_0.go

```diff
@@ -53,145 +51,44 @@ var Upgrade = upgrades.Upgrade{
-		holders          = MAINNET_STNIBI_HOLDERS()              // Pre-upgrade set of holders
-		evmLogs          = []evm.Log{}
-
-		// excessBalance: Send excess ERC20 balance to Nibiru team, so it can be
-		// sent out in case there are new holders before the upgrade.
-		// Gnosis Safe: "cataclysm-1:0x22CBd7CbF3b33681abB3Ced4D64d71acB9a9dCd2"
-		// This helps guarantee the total supply won't change.
-		excessBalance *big.Int
-	)
-
-	// Blank EVM [core.Message] with no value to use as a placeholder for queries
-	// Note that most of these fields are unused when we create EVM instances
-	// outside of an EthereumTx.
-	unusedBigInt := big.NewInt(0)
-	evmMsg := core.Message{
-		To:               &erc20Addr,
-		From:             evm.EVM_MODULE_ADDRESS,
-		Nonce:            nibiru.EvmKeeper.GetAccNonce(ctx, evm.EVM_MODULE_ADDRESS),
-		Value:            unusedBigInt, // amount
-		GasLimit:         0,
-		GasPrice:         unusedBigInt,
-		GasFeeCap:        unusedBigInt,
-		GasTipCap:        unusedBigInt,
-		Data:             []byte{},
-		AccessList:       gethcore.AccessList{},
-		SkipNonceChecks:  false,
-		SkipFromEOACheck: false,
-	}
-	sdb := nibiru.EvmKeeper.NewStateDB(
-		ctx,
-		nibiru.EvmKeeper.TxConfig(ctx, gethcommon.Hash{}),
-	)
-	evmObj := nibiru.EvmKeeper.NewEVM(
-		ctx,
-		evmMsg,
-		nibiru.EvmKeeper.GetEVMConfig(ctx),
-		nil, // tracer: unused
-		sdb,
+		evmLogs  = []evm.Log{}
+		accState = keepers.EvmKeeper.EvmState.AccState
 	)
 
-	totalSupplyErc20, err := nibiru.EvmKeeper.ERC20().TotalSupply(erc20Addr, ctx, evmObj)
-	if err != nil {
-		return err
-	}
-	excessBalance = new(big.Int).Set(totalSupplyErc20)
-	for _, holder := range holders {
-		bal, err := nibiru.EvmKeeper.ERC20().BalanceOf(erc20Addr, holder, ctx, evmObj)
-		if err != nil {
-			return err
-		}
-		holderBalsBefore[holder] = bal
-		excessBalance = new(big.Int).Sub(excessBalance, bal)
-	}
-
 	// -------------------------------------------------------------------------
-	// STEP 2: Erase existing ERC20 contract state
-	// -------------------------------------------------------------------------
-
-	// Now that we have the balances in memory, it's safe to erase the contract
-	// state. The goal is to inject new bytecode and state in.
-	{
-		var erc20StateRange collections.Ranger[evmkeeper.AccStatePrimaryKey] = collections.PairRange[gethcommon.Address, gethcommon.Hash]{}.
-			Prefix(erc20Addr)
-		// ↑ The extra type hint above is meant to make the generics easier to
-		// understand
-
-		iter := nibiru.EvmKeeper.EvmState.AccState.Iterate(ctx, erc20StateRange)
-		defer iter.Close()
-		for ; iter.Valid(); iter.Next() {
-			key := iter.Key()
-			// Deleting through this iterator is safe because the underlying
-			// `sdk.KVStore.Iterator` uses a snapshot. This means deleting the
-			// current key does not affect the iterator unless you call `Set` on that
-			// key again.
-			_ = nibiru.EvmKeeper.EvmState.AccState.Delete(ctx, key)
```

Comment:
We no longer need to delete the old erc20 state anymore, as we're already overwriting only the metadata fields later on.